diff --git a/src/llama-model-loader.h b/src/llama-model-loader.h
index d6b31c231..824bf9f29 100644
--- a/src/llama-model-loader.h
+++ b/src/llama-model-loader.h
@@ -80,6 +80,9 @@ struct llama_model_loader {
     bool check_tensors;
     bool no_alloc;
     bool load_mtp;
+    bool partial_load = false;
+    int32_t linkcpp_layer_begin = -1;
+    int32_t linkcpp_layer_end   = -1;
 
     llama_files files;
     llama_ftype ftype;
@@ -97,6 +100,7 @@ struct llama_model_loader {
     llama_model_set_tensor_data_t set_tensor_data;
     void * set_tensor_data_ud;
     std::vector<ggml_context_ptr> contexts;
+    ggml_context_ptr linkcpp_metadata_ctx;
 
     std::string arch_name;
     LLM_KV      llm_kv    = LLM_KV(LLM_ARCH_UNKNOWN);
